analytics-api/src/analytics_api/config.py

```diff
@@ -0,0 +1,275 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""All of the configuration for the service is captured here.
+
+All items are loaded,
+or have Constants defined here that are loaded into the Flask configuration.
+All modules and lookups get their configuration from the Flask config,
+rather than reading environment variables directly or by accessing this configuration directly.
+"""
+
+import os
+import sys
+
+from dotenv import find_dotenv, load_dotenv
+
+# this will load all the envars from a .env file located in the project root (api)
+load_dotenv(find_dotenv())
+
+
+def get_named_config(config_name: str = 'development'):
+    """Return the configuration object based on the name.
+
+    :raise: KeyError: if an unknown configuration is requested
+    """
+    if config_name in ['production', 'staging', 'default']:
+        config = ProdConfig()
+    elif config_name == 'testing':
+        config = TestConfig()
+    elif config_name == 'development':
+        config = DevConfig()
+    elif config_name == 'docker':
+        config = DockerConfig()
+    else:
+        raise KeyError("Unknown configuration '{config_name}'")
+    return config
+
+
+class _Config():  # pylint: disable=too-few-public-methods
+    """Base class configuration that should set reasonable defaults for all the other configurations."""
+
+    PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+    SECRET_KEY = 'a secret'
+
+    TESTING = False
+    DEBUG = False
+
+    # POSTGRESQL
+    DB_USER = os.getenv('DATABASE_USERNAME', '')
+    DB_PASSWORD = os.getenv('DATABASE_PASSWORD', '')
+    DB_NAME = os.getenv('DATABASE_NAME', '')
+    DB_HOST = os.getenv('DATABASE_HOST', '')
+    DB_PORT = os.getenv('DATABASE_PORT', '5432')
+    SQLALCHEMY_DATABASE_URI = f'postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{int(DB_PORT)}/{DB_NAME}'
+    SQLALCHEMY_BINDS = {
+        'met_db_analytics': f'postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{int(DB_PORT)}/{DB_NAME}'
+    }
+    SQLALCHEMY_ECHO = False
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+    # JWT_OIDC Settings
+    JWT_OIDC_WELL_KNOWN_CONFIG = os.getenv('JWT_OIDC_WELL_KNOWN_CONFIG')
+    JWT_OIDC_ALGORITHMS = os.getenv('JWT_OIDC_ALGORITHMS', 'RS256')
+    JWT_OIDC_JWKS_URI = os.getenv('JWT_OIDC_JWKS_URI')
+    JWT_OIDC_ISSUER = os.getenv('JWT_OIDC_ISSUER')
+    JWT_OIDC_AUDIENCE = os.getenv('JWT_OIDC_AUDIENCE', 'account')
+    JWT_OIDC_CACHING_ENABLED = os.getenv('JWT_OIDC_CACHING_ENABLED', 'True')
+    JWT_OIDC_JWKS_CACHE_TIMEOUT = 300
+
+    S3_BUCKET = os.getenv('S3_BUCKET')
```

Comment:
lets remove all these environment variables not being used (only keep the ones you have in the sample env)